allenact/algorithms/onpolicy_sync/losses/imitation.py

```diff
@@ -62,12 +62,13 @@ def loss(  # type: ignore
             expert_successes = expert_actions_masks.sum()
             should_report_loss = expert_successes.item() != 0
 
-            total_loss = -(
-                expert_actions_masks
-                * actor_critic_output.distributions.log_prob(
-                    cast(torch.LongTensor, expert_actions)
-                )
-            ).sum() / torch.clamp(expert_successes, min=1)
+            log_probs = actor_critic_output.distributions.log_prob(
+                cast(torch.LongTensor, expert_actions)
+            )
+            assert log_probs.shape == expert_actions_masks.shape
```

Comment:
I think this is going to break with the current implementation of the distributions used for navtopartner and the gym_tutorial (I have two log_probs, one for each independent actor in those cases. I could add the log_probs and have only one, but, as of now, that's how it is). Let me know if you want me to update them to work in this (probably more correct) way.

Reply:
I went ahead and fixed this, now it expands the dims of the expert_actions_masks on the right to make number of dimensions equal (to prevent automatic torch broadcast semantics from expanding them on the left by default).